src/main/webapp/3Dscript.js

```diff
@@ -57,143 +88,261 @@ function fetchData() {
   fetch('/getrun?id=' + id + '&dataType=' + type)
       .then((response) => response.json())
       .then((data) => pose = data)
-      .then(() => addMap());
+      .then(() => {
+        const [lat, lng, alt] = findMedians();
+        const latLng = {latitude, longitude};
+        const origin = new Point(lat, lng, alt);
+        addMap(latLng);
+        const [xAxis, yAxis, zAxis] = createInstances();
+        const orientation = {x: xAxis, y: yAxis, z: zAxis};
+        plotTrajectory(origin);
+        plotOrientation(orientation, origin);
+      });
+}
+
+/**
+ * Finds the median value of the latitude, longitude,
+ * and altitude. The median is used as a reference point
+ * for the East-North-Up (ENU) conversion and as the center
+ * point for the 2D map.
+ * @return {array} An array containing median lat, lng, and alt
+ *   coordinates.
+ */
+function findMedians() {
+  const arrays = {lat: [], lng: [], alt: []};
+  for (const point of pose) {
+    arrays.lat.push(point.lat);
+    arrays.lng.push(point.lng);
+    arrays.alt.push(point.alt);
+  }
+  const sorted = {
+    lat: [...arrays.lat].sort((a, b) => a - b),
```

Comment:
just a question: this notation looks cool. what does it do?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax

Reply:
Sorts the array given using the compare function given as a parameter. The a - b specifies that it should be in ascending order. Ill add a comment for this too.